src/rdkafka_op.c

```diff
@@ -570,6 +610,23 @@ rd_kafka_resp_err_t rd_kafka_op_err_destroy (rd_kafka_op_t *rko) {
 }
 
 
+/**
+ * Destroys the rko and returns its error object or NULL if no error.
+ */
+rd_kafka_error_t *rd_kafka_op_error_destroy (rd_kafka_op_t *rko) {
+        if (rko) {
+                rd_kafka_error_t *error = rko->rko_error;
+                rko->rko_error = NULL;
+                rd_kafka_op_destroy(rko);
+                return error;
+        }
+
+        return rd_kafka_error_new(
```

Comment:
error_new() takes a format string

Reply:
I don't understand why this is wrong, since this is just a format string without formatting? I will do something more verbose which i hope is what you are expecting, but i don't get why I'm doing it.

Reply:
But you dont know that err2str() will not return something that can be interpreted as a format string, which makes it unsafe.